apps/jubmoji-quest/src/pages/quests/[id].tsx

```diff
@@ -161,14 +157,24 @@ export default function QuestDetailPage() {
 
   const showLeaderboard = quest.proofType === $Enums.ProofType.TEAM_LEADERBOARD;
 
-  const collectionCardIndices = quest.collectionCards.map((card) => card.index);
-
-  const collectedItems =
-    jubmojis?.filter((jubmoji) =>
-      collectionCardIndices.includes(jubmoji.pubKeyIndex)
-    ).length ?? 0;
+  const collectionCardIndices = quest.powers
+    .flatMap((power) => power.collectionCards.map((card) => card.index))
+    .reduce((unique: number[], index: number) => {
+      if (!unique.includes(index)) {
+        unique.push(index);
+      }
+      return unique;
+    }, []);
 
-  const collectionTotalItems = quest.collectionCards.length;
+  const numPowersCompleted = quest.powers.filter((power) =>
```

Comment:
Same comment here as before; need to check the power's rules before computing "numPowersCompleted"